### ISSUE TITLE
Make attributes searchable in Application and Gateway overview

### ISSUE THREAD

**Issue by laurensslats** [TheThingsNetwork/lorawan-stack#2586]
#### Summary
When attributes are added to Gateways are Application, make the fields searchable in the application and gateway overview page.

#### Why do we need this?
Requested feature.
Making it easier to manage large numbers of devices and gateways.

#### What is already there? What do you see now?
Possibility to create attributes

![Screenshot 2020-05-25 at 16 11 23](https://user-images.githubusercontent.com/17290639/82820435-726f2a80-9ea2-11ea-8d19-26339392f47f.png)

#### What is missing? What do you want to see?
Make it searchable
![Screenshot 2020-05-25 at 16 12 20](https://user-images.githubusercontent.com/17290639/82820483-8dda3580-9ea2-11ea-90ee-0fdc460868b1.png)

#### Environment
v3.8.1+

#### How do you propose to test this?
Test is in staging environment

**Comment by kschiffer** [comment]
Putting this into the July milestone because it's a low hanging fruit and we should provide this feature soon to give more value to entity attributes.

**Comment by pgalic96** [comment]
Just to be sure; This issue is about querying attributes in the attribute list, right? Not searching in the application / device list. If it's the former, right now we are not displaying a list of attributes anywhere. Initially I made a PR to display the attribute list in entity overview, but we dropped it because there wasn't any use for it.

I could implement this search option in the general settings page where the attributes are set, but this seems a bit strange, having a search functionality in the same view as the add/delete functionality. Should we now add the attribute list to the entity overview pages?  cc @kschiffer @bafonins @laurensslats

**Comment by laurensslats** [comment]
I was thinking to include it in the Gateway, application
<img width="378" alt="Screenshot 2020-07-03 at 15 50 31" src="https://user-images.githubusercontent.com/17290639/86476008-00fa8580-bd46-11ea-8a38-02260749eb7c.png">
 & device search bar.

<img width="353" alt="Screenshot 2020-07-03 at 15 50 38" src="https://user-images.githubusercontent.com/17290639/86475576-3fdc0b80-bd45-11ea-82ea-71ee63f686b3.png">

<img width="488" alt="Screenshot 2020-07-03 at 15 52 27" src="https://user-images.githubusercontent.com/17290639/86475578-4074a200-bd45-11ea-9470-7a0620504e72.png">

Attributes can be used to categorize devices and gateways. I.e. adding the attribute `customer` with values `x`, `y`, or `z`. This way, it becomes much easier to find all gateways that are deployed at customer `y`. 

Also, attributes need to be shown. I would suggest to add it to the general overview page of a gateway, application and device.

**Comment by pgalic96** [comment]
Alright, that makes sense.

**Comment by bafonins** [comment]
References https://github.com/TheThingsNetwork/lorawan-stack/issues/1825

**Comment by bafonins** [comment]
@pgalic96 How do you plan to implement this?

IMO, ideally we would have something similar to how search functionality is implemented here on github (searching for issues/prs/etc), but it requires some design work and out current backend doesnt support this.

For now, we can simulate the fuzzy search behavior, but first we need support for `or` when adding `{field}_contains` and support for `attribute_contains`. Which doesnt seem to be implemented. cc @htdvisser any suggestions here?

**Comment by htdvisser** [comment]
Our search API already has support for searching for attributes: the `attributes_contain` field.

It works as follows: If an entity has attribute pairs 

```
attributes: {
  sensor_type: temperature,humidity
  floor: 12
  office: 123
}
```

then it would be matched by a search for

```
attributes_contain: {
  sensor_type: temp
  floor: 12
}
```

**Comment by kschiffer** [comment]
Making the search available as part of the normal search bar of the application/gateway overview is indeed the way to go. Later on, we should look into designing [a dedicated search page](https://github.com/TheThingsNetwork/lorawan-stack/issues/1825) with more sophisticated mechanics, as noted by @bafonins.

It's a bit problematic that the attributes itself are not displayed in the entity table, so it can potentially confusing understanding why a certain entity was matched by the search. However, for the sake of moving forward with this quickly, I think we have to live with that until we look deeper into #1825.

**Comment by kschiffer** [comment]
> For now, we can simulate the fuzzy search behavior, but first we need support for `or` when adding `{field}_contains` and support for `attribute_contains`. Which doesnt seem to be implemented. cc @htdvisser any suggestions here?

This is true, @htdvisser , we can search by attributes, but what we need here is combining different search criteria, which is what we'd need to implement this feature. Ideally, we should have a fuzzy search that basically looks for the search term in all possible columns.

**Comment by htdvisser** [comment]
At the moment all search is done on the database, which doesn't have advanced search features. We would need a real search engine (ElasticSearch) if we want fuzzy search, and that's not something we can quickly implement.

**Comment by pgalic96** [comment]
This has to wait for a better search in the backend, otherwise a confusing search will only annoy users.

**Comment by pgalic96** [comment]
Closed because separate issues are merged together #1825
